### PR TITLE
Updates APM OOTB transaction job docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR updates the documentation of APM transaction jobs to reflect the changes introduced by https://github.com/elastic/kibana/pull/111957

### Preview

[OOTB AD jobs – APM tranasactions](https://stack-docs_1842.docs-preview.app.elstc.co/guide/en/machine-learning/master/ootb-ml-jobs-apm.html#apm-transaction-jobs)

### CODE REVIEW

File: docs/en/stack/ml/anomaly-detection/ootb-ml-jobs-apm.asciidoc [hunk 1]
```diff
@@ -80,15 +80,17 @@ This job is useful in identifying bots.
 [[apm-transaction-jobs]]
 == Transactions
 // tag::apm-transaction-jobs[]
-Detect anomalies in transactions from your APM services. These configurations 
-are only available if data exists that matches the following term query: 
-`"processor.event": "transaction"`.
+Detect anomalies in transactions from your APM services for metric data. These 
+configurations are only available if data exists that matches the following term 
+queries: `"processor.event": "metric"` and `"metrcset.name": "transaction"`.
```
Comment (peteharverson) [inline docs/en/stack/ml/anomaly-detection/ootb-ml-jobs-apm.asciidoc:85]: Typo - should be `metricset.name`, and with the code styling, I don't think you need the quotes `"` around the field names and values here.

Or alternatively, do we want to follow the text used elsewhere `These configurations are only available if data exists that matches the recognizer query specified in the manifest file.` and leave the details for where you describe the job below? The link for the manifest won't be available until https://github.com/elastic/kibana/pull/111957 is merged, but would be `https://github.com/elastic/kibana/blob/master/x-pack/plugins/ml/server/models/data_recognizer/modules/apm_transaction/manifest.json#L8`

File: docs/en/stack/ml/anomaly-detection/ootb-ml-jobs-apm.asciidoc [hunk 2]
```diff
@@ -80,15 +80,17 @@ This job is useful in identifying bots.
...
 
-high_mean_transaction_duration::
+apm_metrics::
 
-* For transaction data where `processor.event` is `transaction`.
-* Models duration of transactions by transaction type for APM services.
-* Detects anomalies in high mean of transaction duration.
+* For metric data where `processor.event` is `metric`.
```
Comment (peteharverson) [inline docs/en/stack/ml/anomaly-detection/ootb-ml-jobs-apm.asciidoc:89]: For completeness, you should say where `processor.event` is `metric` and `metricset.name` is `transaction`

File: docs/en/stack/ml/anomaly-detection/ootb-ml-jobs-apm.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/en/stack/ml/anomaly-detection/ootb-ml-jobs-apm.asciidoc]: Changed in c348ee5.

File: docs/en/stack/ml/anomaly-detection/ootb-ml-jobs-apm.asciidoc (hunk 2 above)
Comment (szabosteve) [reply docs/en/stack/ml/anomaly-detection/ootb-ml-jobs-apm.asciidoc]: Completed in c348ee5.

**Review by peteharverson — APPROVED** [review]
Latest changes LGTM

**Pull request merged**